botocore/handlers.py

```diff
@@ -275,8 +276,16 @@ def signature_overrides(service_data, service_name, session, **kwargs):
         logger.debug("Switching signature version for service %s "
                      "to version %s based on config file override.",
                      service_name, signature_version_override)
-        service_metadata = service_data['metadata']
-        service_metadata['signatureVersion'] = signature_version_override
+        return signature_version_override
+
+
+def disable_signing(**kwargs):
+    """
+    This handler disables request signing by setting the signer
+    name to an empty string, similar to how the signature
+    overrides above work.
+    """
+    return ''
```

Comment:
It's still checking against `None` here:

https://github.com/danielgtaylor/botocore/blob/auth-refactor/botocore/signers.py#L79-L84

Can you elaborate a bit on what you changed?